doc/feature/compartmentReport.md

```diff
@@ -0,0 +1,422 @@
+         *
+         * Preconditions:
+         * - The report was open in write mode.
+         * - writeHeader has already been called.
+         *
+         * @param gid the GID of the cell
+         * @param counts the number of compartments per section
+         * @throw std::runtime_error if writing was not successful.
+         */
+        void writeMapping( const uint32_t gid,
+                           const uint16_ts& counts );
+
+        /** Write the compartment counts for each section for one cell.
+         *
+         * Preconditions:
+         * - The report was open in write mode.
+         * - writeMapping has already been called.
+         *
+         * @param gid the GID of the cell
+         * @param voltages the voltages per compartment to save
+         * @throw std::runtime_error if writing was not successful.
+         */
+        void writeCell( const uint32_t gid, const floats& voltages );
+
+        /** Finish writing the current frame and advance to the next one.
+         *
+         * Upon return getCurrentTime() has been increased to the nearest
+         * multiple of the timestep greater than the value before.
+         */
+        void endFrame();
+    }
+
+## Implementation
+
+This section will cover some implementation details related to the stream
+case as the file based IO is straight forward.
+
+The most two important decisions taken are:
+- The granularity of the messages sent won't be the whole frame but a cell.
+- The mapping will be always included as part of the frame.
+
+The rationale for these decisions follows.
+
+The current API for writing is already cell oriented, while this is convenient
+to write on memory mapped files or H5 files without requiring the client to
+assemble the whole buffer before writing, it's not consistent with the fact that
+reads are whole frame oriented. This means that in the stream case we have to
+assemble the frame at the writer or reader part of the implementation, or
+change the API and force the client to do it on the writer side. Requiring
+the client to assemble the frame is the easiest solution for the implementation
+of the stream based case but it requires bigger changes to adapt existing code
+both in the client side and in the implementation of existing plugins.
+Therefore, the decision here is to keep the API similar to the current one
+and send one message per cell. This allows interesting things like filtering
+cell subsets on origin based on topics, something much more difficult to do
+with whole frames. It also allows very easily having multiple clients receiving
+different portions of the circuit and only those. Obviosly this introduces an
+overhead, but we expect it to be small in relative and absolute terms. At the
+same time we don't expect to stream full compartment data for detailed
+morphologies for more than 10^5 because for larger circuits volumetric
+approaches with voxelization at origin are more reasonable. Having a separate
```

Comment:
Note that "at origin" might mean "in different processes and nodes in the same supercomputer" due to memory pressure in the simulation.